pkg/ctrl/controller.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ctrl
+
+import (
+	"github.com/kubernetes-sigs/kubebuilder/pkg/ctrl/eventhandler"
+	"github.com/kubernetes-sigs/kubebuilder/pkg/ctrl/predicate"
+	"github.com/kubernetes-sigs/kubebuilder/pkg/ctrl/reconcile"
+	"github.com/kubernetes-sigs/kubebuilder/pkg/ctrl/source"
+)
+
+// Controllers are work queues that watch for changes to objects (i.e. Create / Update / Delete Events) and
+// then make changes to the cluster in response (i.e. Reconcile the object).
+type Controller struct {
+	// Name is the name of the Controller that will appear in tracing, logging and monitoring metrics.
+	// Name is required.
+	Name string
+
+	// Reconcile Reconciles observed and desired object state within a cluster.  Defaults to the DefaultReconcileFunc.
```

Comment:
same here: replace "within a cluster" with something else more generic, or remove it.

Reply:
Done